server/src/main/java/org/opensearch/repositories/RepositoriesService.java

```diff
@@ -689,6 +694,15 @@ public static void validateRepositoryMetadataSettings(
                 "setting " + REMOTE_STORE_INDEX_SHALLOW_COPY.getKey() + " cannot be enabled, as remote store feature is not enabled."
             );
         }
+        // Validation to not allow users to create system repository via put repository call.
+        if (isSystemRepositorySettingPresent(repositoryMetadataSettings)) {
+            throw new RepositoryException(
+                repositoryName,
+                "setting "
+                    + SYSTEM_REPOSITORY_SETTING.getKey()
+                    + " cannot be enabled, as put repository doesn't support creation of system repository."
```

Comment:
Should the message be : `can not provide System repository setting [index.private]; this setting is managed by OpenSearch` ,as this is applicable to update as well .